src/posts/components/PostsList.tsx

```diff
@@ -54,13 +56,33 @@ const PostsList: React.FC = () => {
 
   //handle post deletion
   const handleDelete = async (id: number) => {
-    try {
-      await removePost(id);
-      toast.success("Post deleted successfully!");
-      fetchPosts();
-    } catch (error) {
-      toast.error("Failed to delete the post. Please try again.");
-      console.error("Error deleting post:", error);
+    const result = await Swal.fire({
```

Comment:
**performance tip**: try to use minimum packages, you can use primereact dialog instead of swal t reduce installed packages so reduce bundle size